cypress/support/datasetImportPage.js

```diff
@@ -0,0 +1,66 @@
+import { fillInputWithFixture } from './forms';
+
+export const openImportModal = () => {
+    cy.get('.upload.admin button').click();
+};
+
+const defaultExpectedData = ['Row 1', 'Test 1', 'Row 2', 'Test 2'];
+
+export const importDataset = (
+    filename,
+    mimeType = 'text/csv',
+    expectedData = defaultExpectedData,
+) => {
+    openImportModal();
+    fillInputWithFixture(
+        '.btn-upload-dataset input[type=file]',
+        filename,
+        mimeType,
+    );
+
+    cy.get('tbody').should('have.text', expectedData.join(''));
+};
+
+const waitForModalToFadeOut = () => {
+    cy.wait(500);
```

Comment:
why not test for the modal to disappear: 

cy.get( modalSelector, { timeout: 1000 }).should("not.exist");

Reply:
I'll try that!